File: Top-Interview-Questions-Leetcode/Easy-Collections/dynamic-programming/climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n == 1) return 1;
        else if (n == 2) return 2;
        
        int ways;
        
        for (int i = 3, f = 1, s = 2; i <= n; ++i) {
            ways = f + s;
            f = s;
            s = ways;
        }
        
        return ways;
    }
};

File: Top-Interview-Questions-Leetcode/Easy-Collections/dynamic-programming/climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1:
            return 1
        if n == 2:
            return 2
        
        f, s, ways = 1, 2, 0
        
        for _ in range(3, n + 1):
            ways = f + s
            f, s = s, ways
        
        return ways